data/src/main/java/com/microsoft/azure/kusto/data/ClientImpl.java

```diff
@@ -72,21 +73,39 @@ public KustoOperationResult execute(String database, String command) throws Data
 
     @Override
     public KustoOperationResult execute(String database, String command, ClientRequestProperties properties) throws DataServiceException, DataClientException {
+        String response = executeForJsonResult(database, command, properties);
+
+        String clusterEndpoint = determineClusterEndpoint(command);
+        try {
+            return new KustoOperationResult(response, clusterEndpoint.endsWith("v2/rest/query") ? "v2" : "v1");
+        } catch (KustoServiceError e) {
+            throw new DataClientException(clusterEndpoint, "Error converting json response to KustoOperationResult:" + e.getMessage(), e);
+        }
+    }
+
+    @Override
+    public String executeForJsonResult(String command) throws DataServiceException, DataClientException {
+        return executeForJsonResult(DEFAULT_DATABASE_NAME, command);
+    }
+
+    @Override
+    public String executeForJsonResult(String database, String command) throws DataServiceException, DataClientException {
+        return executeForJsonResult(database, command, null);
+    }
+
+    @Override
+    public String executeForJsonResult(String database, String command, ClientRequestProperties properties) throws DataServiceException, DataClientException {
         // Argument validation:
         if (StringUtils.isAnyEmpty(database, command)) {
-            throw new IllegalArgumentException("database or command are empty");
+            throw new IllegalArgumentException("Database and/or command is empty");
         }
         Long timeoutMs = properties == null ? null : properties.getTimeoutInMilliSec();
 
-        String clusterEndpoint;
-        if (command.startsWith(ADMIN_COMMANDS_PREFIX)) {
-            clusterEndpoint = String.format("%s/%s/rest/mgmt", clusterUrl, MGMT_ENDPOINT_VERSION);
-            if (timeoutMs == null) {
+        String clusterEndpoint = determineClusterEndpoint(command);
+        if (timeoutMs == null) {
+            if (command.startsWith(ADMIN_COMMANDS_PREFIX)) {
```

Comment:
Maybe out of scope for this PR - but maybe we want to trim the command first? 
If someone starts an admin command with a space or tab they would get the wrong results